From d33562db3da2820924a61e356e3dd735ee9ee795 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Tue, 4 Sep 2018 17:19:56 +0900
Subject: [PATCH 365/909] rcar-vin: rcar-csi2: Add blank margin when
 caluculating bit rate

commit fbd82dfccd1c560ab23a7f07b67b4f69e84c4813 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

When csi40 is used in 1 lane, bps becomes a small value and input size
of 720x480@60Hz can not be captured. This patch set to the correct
value by adding a margin at bit rate calculation. By this fix,
720x480@60Hz resolution can be captured in 1 lane of csi40.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/rcar-vin/rcar-csi2.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/media/platform/rcar-vin/rcar-csi2.c b/drivers/media/platform/rcar-vin/rcar-csi2.c
index f5ba9bf0d99f..7180f5da7703 100644
--- a/drivers/media/platform/rcar-vin/rcar-csi2.c
+++ b/drivers/media/platform/rcar-vin/rcar-csi2.c
@@ -456,6 +456,13 @@ static int rcsi2_calc_mbps(struct rcar_csi2 *priv, unsigned int bpp)
 	 * bps = link_freq * 2
 	 */
 	mbps = v4l2_ctrl_g_ctrl_int64(ctrl) * bpp;
+
+	/* Vblank's margin is 1.05 times of the horizontal size */
+	mbps = div_u64(mbps * 105, 100);
+
+	/* Hblank's margin is 1.13 times of the vertical size */
+	mbps = div_u64(mbps * 113, 100);
+
 	do_div(mbps, priv->lanes * 1000000);
 
 	return mbps;
-- 
2.17.1

